libnopegl/src/hmap.c

```diff
@@ -38,13 +38,13 @@ struct hmap {
     size_t size;
     size_t mask;
     size_t count; // total number of entries
-    user_free_func_type user_free_func;
+    hmap_user_free_func_type user_free_func;
```

Comment:
Can we define `ngli_user_free_func_type` (or even ngli_user_freep_func_type if that makes sense , that would in line with our current `ngli_freep_func_type`) in `utils.h` and use it for both hmap and darray ?

Reply:
done